File: backend/src/schema/hub.py
from uuid import UUID
from pydantic import BaseModel, Field


class ResponseHub(BaseModel):
    name: str
    position: tuple[int, int]
    access: str
    color: str | None = None
    drones: list[UUID] = []
    capacity: int = Field(ge=1, default=1)
    connections: list[UUID] = []

File: frontend/src/schemas/hub.ts
import { z } from "zod"

export const HubSchema = z.object({
	name: z.string(),
	position: z.tuple([z.number(), z.number()]),
	access: z.string(),
	color: z.optional(z.string()),
	connections: z.array(z.uuidv4()),
	capacity: z.number(),
	drones: z.array(z.uuidv4()),
})
